public/app/core/components/Select/MetricSelect.tsx

```diff
@@ -7,7 +7,7 @@ import { Variable } from 'app/types/templates';
 const { Select } = LegacyForms;
 
 export interface Props {
-  onChange: (value: string) => void;
+  onChange: (value: string | undefined) => void;
```

Comment:
Could this just be
```suggestion
  onChange: (value?: string) => void;
```

Reply:
no, it's not the same. The argument is not optional, but can be undefined

Reply:
Cool, makes sense